### PR TITLE
Add RemoteLogTracingEvent to allow tracing the event happen or not

### PR DESCRIPTION
**Pull request by chenjian2664**
## Description
Add RemoteTracingEvent to support check a database event(in log) happened or not without grab all logs from container at a time.
Update Postgresql testCancellation test to use RemoteTracingEvent.

## Additional context and related issues

https://trinodb.slack.com/archives/CP1MUNEUX/p1710496005499909

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/RemoteLogTracingEvent.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+public class RemoteLogTracingEvent
+        implements Consumer<RemoteDatabaseEvent>
+{
+    private final Predicate<RemoteDatabaseEvent> condition;
+    private final AtomicBoolean happenStatus;
+
+    public RemoteLogTracingEvent(Predicate<RemoteDatabaseEvent> condition)
+    {
+        this.condition = requireNonNull(condition, "Condition is null");
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/RemoteLogTracingEvent.java:30]: nit:

```suggestion
        this.condition = requireNonNull(condition, "condition is null");
```

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/RemoteLogTracingEvent.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        this.happenStatus = new AtomicBoolean();
+    }
+
+    public boolean happened()
+    {
+        return happenStatus.get();
+    }
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/RemoteLogTracingEvent.java:37]: nit: consider renaming field to `happened` and method to `hasHappened`.

```suggestion
    public boolean hasHappened()
    {
        return happened.get();
    }
```

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java [hunk 3]
```diff
@@ -84,6 +92,59 @@ public TestingPostgreSqlServer(boolean shouldExposeFixedPorts)
...
+        {
+            if (tracingEvents.isEmpty()) {
+                return;
+            }
+
+            buildEvent(outputFrame)
+                    .ifPresent(remoteDatabaseEvent -> tracingEvents.forEach(tracingEvent -> tracingEvent.accept(remoteDatabaseEvent)));
+        }
+
+        private Optional<RemoteDatabaseEvent> buildEvent(OutputFrame outputFrame)
```
Comment (hashhar) [inline plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java:111]: Is `getRemoteDatabaseEvents` now safe to remove?

**Review by hashhar — COMMENTED** [review]
Looks good to me.

Only naming comments.

Also will you remove `getRemoteDatabaseEvents` in follow-ups?

**Comment by chenjian2664** [comment]
> Looks good to me.
> 
> Only naming comments.
> 
> Also will you remove `getRemoteDatabaseEvents` in follow-ups?

Thanks for the review.
Yes I am happy to

File: plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java (hunk 3 above)
Comment (chenjian2664) [reply plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestingPostgreSqlServer.java:111]: There are multiple references of the `getRemoteDatabaseEvents`, I think it's not so easy to remove the method directly.

**Comment by chenjian2664** [comment]
Error:    TestHiveRuntimeAdaptivePartitioningFaultTolerantExecutionJoinQueries.testInnerJoinWithEmptyProbeSide » Timeout testInnerJoinWithEmptyProbeSide() timed out after 120 seconds

**Review by hashhar — APPROVED** [review]
@Praveen2112 / @kokosing do you want to take a look before I merge?

**Review by Praveen2112 — APPROVED** [review]

**Pull request merged**